File: Languages/npeg_java/robusthaven/robusthaven/text/ParsingFatalTerminalException.java
package robusthaven.text;

public class ParsingFatalTerminalException extends RuntimeException {
    long m_errorPos;

    public long getErrorPosistion() {
	return m_errorPos;
    }

    public ParsingFatalTerminalException(String msg, long pos) {
	super(msg);
	m_errorPos = pos;
    }
}

File: Languages/npeg_javascript/ParsingFatalTerminalException.js
if(!RobustHaven) var RobustHaven={};
if(!RobustHaven.Text) RobustHaven.Text={};
if(!RobustHaven.Text.Npeg) RobustHaven.Text.Npeg={};

RobustHaven.Text.Npeg.ParsingFatalTerminalException = function(message, pos) {
    var m_errorPos = pos,
        message = (typeof message == undefined) ? '' : message;

    RobustHaven.Text.Npeg.ParsingFatalTerminalException.superclass.constructor.call(this, message);

    this.getErrorPosition = function() {
        return m_errorPos;
    }
}
RobustHaven.Text.Npeg.extend(
    RobustHaven.Text.Npeg.ParsingFatalTerminalException, RobustHaven.Text.Npeg.ParsingExpressionGrammarException);